pandas/tests/strings/test_find_replace.py

```diff
@@ -175,17 +175,14 @@ def test_contains_na_kwarg_for_nullable_string_dtype(
 
     values = Series(["a", "b", "c", "a", np.nan], dtype=nullable_string_dtype)
 
-    msg = (
-        "Allowing a non-bool 'na' in obj.str.contains is deprecated and "
-        "will raise in a future version"
-    )
-    warn = None
-    if not pd.isna(na) and not isinstance(na, bool):
-        warn = FutureWarning
-    with tm.assert_produces_warning(warn, match=msg):
+    if na in [0, 3] and na is not False:
```

Comment:
I was confused by the `na is not False`, at first. I suppose that is to avoid `False in [0, 3]` to return True? 

I was going to suggest to do `if isinstance(na, int)` instead, but that has the same problem of recognizing bools .. (and then `if isinstance(na, int) and not isinstance(na, bool)` is maybe not that much clearer)

Reply:
That's correct. Happy to make a change if you have a preference, but it seems okay as-is to me.